imbi/automations/pagerduty.py

```diff
@@ -104,3 +104,88 @@ async def _delete_project(context: automations.AutomationContext,
+        return
+
+    if not (dependency_service_id := await _get_service_id(
+            context, dependency.dependency_id)):
+        context.note_progress(
+            'Pagerduty service not set up for '
+            'project %d, skipping automation', dependency.dependency_id)
+        return
+
+    client = await pagerduty.create_client(context.application,
+                                           automation.integration_name)
+
+    service_dependencies = await client.get_service_dependencies(
+        dependency.project_id)
+    for d in service_dependencies:
+        if d.supporting_service.id == dependency_service_id:
+            context.note_progress(
+                'Pagerduty service dependency already exists '
+                'between %s and %s, skipping automation',
+                dependency.project_id, dependency.dependency_id)
+            return
+
+    await client.create_service_dependency(
+        supporting_service_id=dependency_service_id,
+        dependent_service_id=dependent_service_id)
+
+
+async def disassociate_service_dependency(
+    context: automations.AutomationContext,
+    automation: models.Automation,
+    dependency: models.ProjectDependency,
+) -> None:
+    if not (dependent_service_id := await _get_service_id(
+            context, dependency.project_id)):
+        context.note_progress(
+            'Pagerduty service not set up for '
+            'project %d, skipping automation', dependency.project_id)
+        return
+
+    if not (dependency_service_id := await _get_service_id(
+            context, dependency.dependency_id)):
+        context.note_progress(
+            'Pagerduty service not set up for '
+            'project %d, skipping automation', dependency.dependency_id)
+        return
+
+    client = await pagerduty.create_client(context.application,
+                                           automation.integration_name)
+
+    service_dependencies = await client.get_service_dependencies(
+        dependency.project_id)
+    for d in service_dependencies:
+        if d.supporting_service.id == dependency_service_id:
+            await client.remove_service_dependency(
+                supporting_service_id=dependency_service_id,
+                dependent_service_id=dependent_service_id)
+            return
+    context.note_progress(
+        'Pagerduty service dependency does not exist '
+        'between %s and %s, skipping automation', dependency.project_id,
+        dependency.dependency_id)
+
+
+async def _get_service_id(context: automations.AutomationContext,
+                          project_id: int) -> typing.Optional[str]:
+    result = await context.run_query(
+        'SELECT pagerduty_service_id'
+        '  FROM v1.projects'
+        ' WHERE id = %(project_id)s', {'project_id': project_id},
+        'get-pagerduty-service-id')
+    if not result.row_count:
+        return None
+    return result.row['pagerduty_service_id']
```

Comment:
These are stored in [the `v1.project_identifiers` table](https://github.com/AWeber-Imbi/imbi-api/blob/ce6019900fb427bc05be3f966cb6c0bcc795ebc5/imbi/automations/pagerduty.py#L61-L70). You can retrieve them with something like
```python
result = await context.run_query(
    'SELECT external_id'
    '  FROM v1.project_identifiers'
    ' WHERE project_id = %(project_id)s'
    '   AND integration_name = %(integration_name)s', {
        'project_id': project_id,
        'integration_name': automation.integration_name,
    },
    metric_name='get-pagerduty-service-id')
```
I've been meaning to remove the various columns of `v1.projects` that were replaced by the surrogate IDs.

Reply:
Good catch